### PR TITLE
[SYCL][ROCM] Added ROCm backend to device filter

### PR DESCRIPTION
**Pull request by AidanBeltonS**
Adding rocm backend to SyclBeMap, so the device filter does not throw errors with ROCm backend
Updated error message from device filter to include ROCm.

### CODE REVIEW

**Comment by vladimirlaz** [comment]
@AidanBeltonS could you please apply clang-format

File: sycl/source/detail/device_filter.cpp [hunk 1]
```diff
@@ -75,7 +75,7 @@ device_filter::device_filter(const std::string &FilterString) {
     } catch (...) {
       std::string Message =
           std::string("Invalid device filter: ") + FilterString +
-          "\nPossible backend values are {host,opencl,level_zero,cuda,*}.\n"
+          "\nPossible backend values are {host,opencl,level_zero,cuda,rocm,*}.\n"
```
Comment (bader) [inline sycl/source/detail/device_filter.cpp:78]:
```suggestion
          "\nPossible backend values are"
          "{host,opencl,level_zero,cuda,rocm,*}.\n"
```

Comment (AidanBeltonS) [reply sycl/source/detail/device_filter.cpp]: Updated branch with clang format change.
Let me know if there is anything else you wish to rectify.
Thanks

**Review by vladimirlaz — APPROVED** [review]

**Comment by bader** [comment]
@AidanBeltonS, please, update the unit test.

**Review by bader — APPROVED** [review]

**Review by vladimirlaz — APPROVED** [review]

**Pull request merged**